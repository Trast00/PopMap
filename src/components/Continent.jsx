import React from 'react'
import { useSelector } from 'react-redux'
import { listContinents } from '../redux/country/countryReducer'
import Country from './ListContry/Country'
import { FaArrowCircleLeft, FaArrowCircleRight } from "react-icons/fa";


const Continent = () => {
  const indexCurrentContinents = useSelector(state => state.countryReducer.indexCurrentContinents)
  return (
    <div className='max-width flex-center continent'>
      <FaArrowCircleLeft className='icons filter-back' />
      {(indexCurrentContinents!==1) 
      && <Country country={listContinents[0]} detailed={true} />
      }  
      <FaArrowCircleRight className='icons filter-next' />
    </div>
  )
}

export default Continent
