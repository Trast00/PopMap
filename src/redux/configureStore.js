import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countryReducer';

export default configureStore({
  reducer: {
    countryReducer,
  },
});
