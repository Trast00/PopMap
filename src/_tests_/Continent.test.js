import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import Continent from '../components/Continent';
import configureStore from '../redux/configureStore';

describe('test render', () => {
  test('Conutry should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={configureStore}>
          <Continent />
        </Provider>
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
