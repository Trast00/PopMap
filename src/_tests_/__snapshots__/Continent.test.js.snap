// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Conutry should match snapshoot 1`] = `
<div
  className="max-width flex-center continent"
>
  <div
    className="flex-center column country detail-country"
  >
    <img
      alt="WorldCountry Images"
      src="world.svg"
    />
    <div
      className="column detail-wrapper"
    >
      <h4>
        World
      </h4>
      <p>
        Citizens: 
         
        0
      </p>
    </div>
  </div>
  <svg
    className="icons filter-next"
    fill="currentColor"
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
    />
  </svg>
</div>
`;
